lib/ae_page_objects/concerns/visitable.rb

```diff
@@ -24,7 +24,12 @@ module VisitMethod
         def visit(*args)
           raise ArgumentError, "Cannot pass block to visit()" if block_given?
 
-          full_path = site.generate_path(paths.first, *args)
+          args = args.dup
+          inner_options = args.last.is_a?(::Hash)? args.pop : {}
+
+          path = inner_options.delete(:via) || paths.first
+
+          full_path = site.generate_path(path, args, inner_options)
```

Comment:
`generate_path` should receive the path and the exact same `args` as passed to `visit` except for the `:via` key when present. 

Please add tests covering the following invocations:

```ruby
PageObjects::Books::ShowPage.visit(book, :format => :json)
# should invoke generate_path(<path>, book, :format => :json)

PageObjects::Books::ShowPage.visit(book, :format => :json, :via => :something)
# should invoke generate_path(<path>, book, :format => :json)

PageObjects::Books::ShowPage.visit(something, something_else)
# should invoke generate_path(<path>, something, something_else)

PageObjects::Books::ShowPage.visit(something, something_else, :via => "something")
# should invoke generate_path(<path>, something, something_else)
```